### ISSUE TITLE
Switching between sessions clears harpoon entries

### ISSUE THREAD

**Issue by sevi-kun** [ThePrimeagen/harpoon#594]
Yes, this is harpoon2.

So, I use the   session-lens and auto-session plugins from [rmagatti](https://github.com/rmagatti) to switch between my projects. But when doing so, all entries of harpoon are being cleared. 

My neovim configuration is available here: [https://github.com/sevi-kun/nvim](https://github.com/sevi-kun/nvim)

**Comment by maxrzaw** [comment]
This could be due to your cwd changing. Changing the key function in the harpoon settings might help. The default is `vim.uv.cwd()`.

**Comment by sevi-kun** [comment]
This could be it. I will look into it as soon as I have a bit more time.

**Comment by maxrzaw** [comment]
> This could be it. I will look into it as soon as I have a bit more time.

Unfortunately, custom keys are broken right now. Which might actually be the root cause for your issue. Might be worth trying the branch from #572

**Comment by sevi-kun** [comment]
No... I tried the pr but my harpoon entries are still completely empty after switching sessions. Maybe the sessions plugin actually removes some entries? I hope I can look into the source code soon.

**Comment by simaxme** [comment]
I also have a problem that may be associated with this issue:
- Using [possession.nvim](https://github.com/jedrzejboczar/possession.nvim) the harpoon list does not restore. 
- When opening the project while being outside of the project folder through the nvim path argument, it also does not restore.
The only way it works is by manually jumping into the folder and typing `nvim .`

**Comment by 2elli** [comment]
Is there any update or solution to this?

I'm running into this issue as well using [persistence.nvim](https://github.com/folke/persistence.nvim). When restoring a session or switching to a new session it does not restore the saved list.
Works as intended if I cd to the project directory and open with `nvim .`
My [config](https://github.com/2elli/dots/tree/master/dot_config/nvim).

**Comment by searleser97** [comment]
same issue here, is there a way to save the harpoon list in a session file @ThePrimeagen  ? already tried changing the key and it didn't help

**Comment by searleser97** [comment]
after taking a look at the code and understanding how harpoon is working, what worked for me was delaying the initialization of harpoon until the directory returned by `vim.loop.cwd()` is correctly updated by nvim-rooter (which in my case, happens after I select the session I want to open)
